### PR TITLE
IBC v1.0.0

### PR DESCRIPTION
**Pull request by fedekunze**
replaces #130

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: ibc-demo.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
+# IBC instruction
+
+## Dependencies
+
+This branch uses non-canonical branch of cosmos-sdk. Before building, run `go mod vendor` on the root directory to retrieve the dependencies. To build:
+
+```shell
+git clone git@github.com:cosmos/gaia
+cd gaia
+git checkout cwgoes/ibc-demo-fixes
```
Comment (jackzampolin) [inline ibc-demo.md]:
```suggestion
git checkout fedekunze/ibc
```

File: ibc-demo.sh [hunk 2]
```diff
@@ -0,0 +1,178 @@
+#!/bin/zsh
+
+GAIA_BRANCH=cwgoes/ibc-demo-fixes
```
Comment (jackzampolin) [inline ibc-demo.sh]:
```suggestion
GAIA_BRANCH=fedekunze/ibc
```

File: ibc-demo.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+gaiad testnet -o ibc1 --v 1 --chain-id ibc1 --node-dir-prefix n
+```
+
+### Set `gaiad` and `gaiacli` Configuation
+
+Fix the configuration files for both `gaiad` and `gaiacli` to allow both chains/nodes to run on the same machine:
+
+```bash
+# Configure the proper database backend for each node and different listening ports
+sed -i 's/"leveldb"/"goleveldb"/g' ibc0/n0/gaiad/config/config.toml
```
Comment (ancazamfir) [inline ibc-demo.md]: To work fine also for MAC (BSD sed) users:
```suggestion
sed -i'.orig' -e 's/"leveldb"/"goleveldb"/g' ibc0/n0/gaiad/config/config.toml
```

File: ibc-demo.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+ibc0 <- update_client           [OK] txid(FA22E93601218CEA839FDEB7BD0D8F47D81E1172E18A8B21717675FF5C4BCF40) client(ibconeclient)
+ibc0 <- channel_open_ack        [OK] txid(1840343AFB2D5666F52440C199A1356C63A884A959F0A9A53175773CDB83006B) portid(bankbankbank) chanid(channelzero)
+ibc1 <- update_client           [OK] txid(BBE212C5041AC366C018BB97F8DF8A495562EAFFFF51BFDA7BBAE9952BE589D0) client(ibczeroclient)
+ibc1 <- channel_open_confirm    [OK] txid(69F50CA44AE6AD84BD24866E7DB7FE8ADFD9C484171662CB9E6F0C71BFC222A9) portid(bankbankbank) chanid(channelone)
+```
+
+You can query the `channel` after establishment by running the following command:
+
+```bash
+gaiacli --home ibc0/n0/gaiacli q ibc channel end bankbankbank channelzero --indent --trust-node
```
Comment (ancazamfir) [inline ibc-demo.md]:
```suggestion
gaiacli --home ibc0/n0/gaiacli q ibc channel end bank channelzero --indent --trust-node
```

File: ibc-demo.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+gaiacli --home ibc1/n0/gaiacli q ibc channel end bankbankbank channelone --indent --trust-node
```
Comment (ancazamfir) [inline ibc-demo.md]:
```suggestion
gaiacli --home ibc1/n0/gaiacli q ibc channel end bank channelone --indent --trust-node
```

File: ibc-demo.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+> NOTE: This commands returns the `height` at which it was committed, this should be at the beginning of the JSON output. The enviornment variable `TIMEOUT`.
+
+### Receive Packet
+
+Now, try querying the account on `ibc1` that you sent the `1stake` to, the account will be empty:
+
+```bash
+gaiacli --home ibc1/n0/gaiacli q account $(gaiacli --home ibc0/n0/gaiacli keys show n1 -a)
```
Comment (ancazamfir) [inline ibc-demo.md]:
```suggestion
gaiacli --home ibc1/n0/gaiacli q account $(gaiacli --home ibc0/n0/gaiacli keys show n1 -a) --indent --trust-node
```

**Review by ancazamfir — COMMENTED** [review]
Ran the demo, good stuff! A few minor comments.

File: ibc-demo.md (hunk 3 above)
Comment (ancazamfir) [reply ibc-demo.md]: Apply also to the other `sed -i ...` below.

**Comment by fedekunze** [comment]
replaced by #288

**Pull request closed without merging**